core/challenges.go

```diff
@@ -5,31 +5,73 @@
 
 package core
 
+import (
+	"encoding/json"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/letsencrypt/go-jose"
+)
+
+func newChallenge(challengeType string, accountKey *jose.JsonWebKey) (Challenge, error) {
+	ak := AuthorizedKey{
+		Token: NewToken(),
+		Key:   accountKey,
+	}
+
+	jsonAK, err := json.Marshal(ak)
```

Comment:
You didn't respond to my comment about this marshalling the AK three times in ChallengesFor. I asked: why do we needed to serialize this here, instead of at the time the Challenge gets serialized?